tests-integration/common.go

```diff
@@ -23,29 +21,63 @@ func findOpenTCPPort(t *testing.T) int {
 
 func assertLineprotocolEqual(t *testing.T, expect, got string) bool {
 	t.Helper()
-	return assert.Equal(t, cleanupLP(expect), cleanupLP(got))
-}
 
-func cleanupLP(s string) []string {
-	lines := strings.Split(s, "\n")
-	var cleanLines []string
-	for _, line := range lines {
-		line = strings.TrimSpace(line)
-		if len(line) == 0 {
-			continue
-		}
-		cleanLines = append(cleanLines, sortFields(line))
+	expectedPoints, err := parseLineProtocol(expect)
+	if err != nil {
+		t.Error(err)
+	}
+
+	actualPoints, err := parseLineProtocol(got)
+	if err != nil {
+		t.Error(err)
+	}
+
+	sameLength := assert.Len(t, actualPoints, len(expectedPoints))
+	if !sameLength {
+		return sameLength
+	}
+
+	// order of LP within the batch is not guaranteed, so we cannot do pairwise comparison
+	// instead, we create a map of Point::HashID() -> Point and compare the two maps
+
+	expectedMap := pointMap(expectedPoints)
+	actualMap := pointMap(actualPoints)
+
+	equality := make(map[uint64]bool, len(expectedMap))
+	for k, v := range expectedMap {
+		equality[k] = assertPointEqual(t, v, actualMap[k])
 	}
-	sort.Strings(cleanLines)
-	return cleanLines
+
+	// assert all true
+	return assert.NotContains(t, equality, false)
 }
 
-func sortFields(line string) string {
-	fieldsIndexes := regexp.MustCompile(`^\s*(\S+)\s+(\S+)\s*(\d*)\s*$`).FindStringSubmatchIndex(line)
-	if len(fieldsIndexes) != 8 {
-		panic(fmt.Sprint(len(fieldsIndexes), line))
+func pointMap(points []models.Point) map[uint64]models.Point {
```

Comment:
Adding `influxdata/influxdb` as a dependency to any project is often problematic. Also, I deliberately made the interface for this repository not include dependencies on `influxdata/influxdb` or `influxdb/line-protocol` because those can easily be added by the user of the repo.

In this test-only module, it's fine, but I do wonder what the benefit of `models.Point` is, and whether we should just copy the struct into this repo?
https://go-proverbs.github.io/

Reply:
Eh, I tried and gave up.

Reply:
Hehe.